### ISSUE TITLE
`ng build my-library` crashes after migrating from Angular 12 to 13

### ISSUE THREAD

**Issue by PowerKiKi** [ng-packagr/ng-packagr#2261]
## Type of Issue

```
[x] Bug Report
[ ] Feature Request
```

## Description

I created a project with angular-cli that includes a library a long time ago with Angular 7. After migrating from Angular 12 to 13, it can no longer build and it crashes immediately like so:

```sh
$ ng build my-library

No provider for InjectionToken ng.v5.defaultTsConfig! (InjectionToken ng.v5.packageTransform -> InjectionToken ng.v5.initTsConfigTransform -> InjectionToken ng.v5.defaultTsConfig)
```

#### How To Reproduce

1.
   ```sh
   ng new demo --no-routing --style scss
   cd demo/
   ng generate library my-library
   ```
2. Remove `defaultConfiguration` like so:
    ```diff
    --- angular.json
    +++ angular.json
    @@ -124,8 +124,7 @@
                 "development": {
                   "tsConfig": "projects/my-library/tsconfig.lib.json"
                 }
    -          },
    -          "defaultConfiguration": "production"
    +          }
             },
             "test": {
               "builder": "@angular-devkit/build-angular:karma",
    ```

#### Expected Behaviour

Build without crashes.

#### Solution

Manually add the missing `defaultConfiguration` in old projects.

Since there is a clear and permanent solution, I am not too sure if `ng-packagr` should be fixed or not. From my point of view the problem is solved, but maybe `ng-packagr` could handle the case in a better way ? by having a better error message ? or defaulting on production when the value is missing ?

Feel free to close the issue as wontfix if maintainers feel like it's not worth their time.

#### Version Information

```
$ node_modules/.bin/ng-packagr --version
ng-packagr:            13.2.1
@angular/compiler:     13.2.4
rollup:                2.68.0
typescript:            4.5.5
```

**Comment by rbirkgit** [comment]
I get this error too after ang13 update in a lib that doesn't have a tsconfig. I assume you get this error because without the default configuration, you don't specify a tsconfig either.

**Comment by diegofelipece** [comment]
Same problem here

**Comment by hackmajoris** [comment]
Don't know your context, but I solved the issue by providing this congifs in project.json:

```
 "configurations": {
        "production": {
          "tsConfig": "libs/shared/angular-components/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "libs/shared/angular-components/tsconfig.lib.json"
        }
  },
```

**Comment by JoostK** [comment]
This was resolved in #2646
